gateway-js/src/utilities/waitUntil.ts

```diff
@@ -0,0 +1,18 @@
+export function waitUntil<T = void>() {
```

Comment:
Is this new in this PR or brought in from somewhere else?

It looks fine, but FWIW I've been using https://www.npmjs.com/package/@josephg/resolvable in apollo-server, which is just a Promise with resolve and reject methods on it.  This waitUntil approach is better if the Promise is actually exposed "publicly" but when it's just used within a test or in very internal state, the `resolvable` approach where the resolve lives directly on the object it controls seems a little conceptually simpler?

Reply:
Just came up with it, found it a bit nicer to use than `new Promise`. Like `resolvable` even more though!
06054070